### PR TITLE
1672: Carousel makeover

### PR DESCRIPTION
**Pull request by xendk**
Major rewrite of carousel, including: multiple carousels, pre-loaded
tabs, progressive loading, pre-loading of inactive tabs, tab switching
transitions, progressive fetching of covers from backend.

Requires https://github.com/ding2/ding2/pull/195

### CODE REVIEW

File: modules/ting_search_carousel/ting_search_carousel.pages.inc [hunk 1]
```diff
@@ -7,33 +7,25 @@
...
- * @param array $index
- *   Array index storing specific data.
+ * @param int $index
+ *   Configuration index.
+ * @param int $offset
+ *   Offset into result to fetch from.
  */
-function ting_search_carousel_result($index) {
+function ting_search_carousel_result($index, $offset) {
   $content = FALSE;
```
Comment (JrgenGNielsen) [inline modules/ting_search_carousel/ting_search_carousel.pages.inc]: leftover code? $content is defined as an empty string two lines down

**Comment by JrgenGNielsen** [comment]
apart from a small detail: top nice! reviewed and accepted

File: modules/ting_search_carousel/ting_search_carousel.pages.inc (hunk 1 above)
Comment (xendk) [inline modules/ting_search_carousel/ting_search_carousel.pages.inc]: Quite. I'll remove it.

**Comment by xendk** [comment]
@JrgenGNielsen I've added 3 small cleanup commits. No functionality change, just changing the field description to not refer to ADDI, logging as ting_cover rather than ting_cover_addi, and obsoleted file references in the info file.

**Comment by xendk** [comment]
@JrgenGNielsen I've added two more cleanup commits. One prints out the description for the first tab, and the second removes TRUE/FALSE as a class.

**Comment by cableman** [comment]
Merged by hand to resolve conflict in compiled CSS.

**Pull request merged**